pkg/cloud/azure/actuators/machine/conditions.go

```diff
@@ -0,0 +1,89 @@
+package machine
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
+)
+
+const (
+	// MachineCreationSucceeded indicates success for machine creation
+	MachineCreationSucceeded = "MachineCreationSucceeded"
+
+	// MachineCreationFailed indicates that machine creation failed
+	MachineCreationFailed = "MachineCreationFailed"
+)
+
+func shouldUpdateCondition(
+	oldStatus corev1.ConditionStatus, oldReason, oldMessage string,
```

Comment:
we could pass the object oldCondition/newCondition here which is semantically more appropriate for the func name? e.g https://github.com/openshift/cluster-api-provider-gcp/blob/master/pkg/cloud/gcp/actuators/machine/conditions.go#L15-L17

Can we put some unit https://github.com/openshift/cluster-api-provider-gcp/blob/master/pkg/cloud/gcp/actuators/machine/conditions_test.go